src/rules/no_dupe_keys.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright 2020 the Deno authors. All rights reserved. MIT license.
+use super::Context;
+use super::LintRule;
+use std::collections::{BTreeSet, HashSet};
+use swc_atoms::JsWord;
+use swc_ecma_ast::Prop;
+use swc_ecma_ast::Prop::*;
+use swc_ecma_ast::PropName;
+use swc_ecma_ast::PropName::*;
+use swc_ecma_ast::PropOrSpread::{Prop as PropVariant, Spread};
+use swc_ecma_ast::{Module, ObjectLit, PropOrSpread};
+use swc_ecma_visit::{Node, Visit};
+
+pub struct NoDupeKeys;
+
+impl LintRule for NoDupeKeys {
+  fn new() -> Box<Self> {
+    Box::new(NoDupeKeys)
+  }
+
+  fn lint_module(&self, context: Context, module: Module) {
+    let mut visitor = NoDupeKeysVisitor::new(context);
+    visitor.visit_module(&module, &module);
+  }
+}
+
+pub struct NoDupeKeysVisitor {
+  context: Context,
+}
+
+impl NoDupeKeysVisitor {
+  pub fn new(context: Context) -> Self {
+    Self { context }
+  }
+}
+
+impl Visit for NoDupeKeysVisitor {
+  fn visit_object_lit(&mut self, obj_lit: &ObjectLit, _parent: &dyn Node) {
+    let mut keys: HashSet<JsWord> = HashSet::new();
+    let mut duplicates: BTreeSet<JsWord> = BTreeSet::new();
```

Comment:
Nitpick - use `String` instead of `JsWord`

Reply:
No problem, I'll put the change in now.

For my own understanding, what's the benefit of `String` over `JsWord`?

Reply:
Purely cosmetic; one less foreign type that needs to be looked up